R/utils.R

```diff
@@ -167,34 +167,70 @@ repair_variable_names <- function(names) {
   names
 }
 
-
-
-
 #' Dump data to temporary file in format readable by CmdStan
 #'
-#' Currently calls `rstan::stan_rdump()` to create the `.data.R` file.
-#' FIXME:
-#'
 #' @param data A named list of \R objects.
 #' @return Path to temporary file containing the data.
 #' @noRd
-write_rdump <- function(data) {
-  # FIXME don't use rstan
-  if (!requireNamespace("rstan", quietly = TRUE)) {
-    stop("Please install the 'rstan' package. This is temporarily required ",
-         "for writing the data file for CmdStan.", call. = FALSE)
+write_json <- function(data) {
+  if (!requireNamespace("jsonlite", quietly = TRUE)) {
+    stop("Please install the 'jsonlite' package.", call. = FALSE)
   }
   checkmate::assert_names(names(data), type = "unique")
-  temp_file <- tempfile(pattern = "standata-", fileext = ".data.R")
-  rstan::stan_rdump(
+  temp_file <- tempfile(pattern = "standata-", fileext = ".json")
+  cmdstanr_jsondump(
     list = names(data),
     file = temp_file,
     envir = as.environment(data)
   )
   temp_file
 }
 
-
+#' Dump data to temporary file in format readable by CmdStan
+#'
+#' @param list A vector of character string: the names of one or more R objects to be dumped.
+#' @param file A character string represeneting the path to the output file
+#' @param envir The environment to search for the listed objects.
+#' @export
+#' @return Path to temporary file containing the data.
+cmdstanr_jsondump <- function(list,
+                          file,
+                          envir = parent.frame()) {
```

Comment:
Also, for the exported function what do you think of the name `write_stan_json()` instead of `cmdstanr_jsondump()`? I don't have a strong preference so up to you, just throwing it out there as an option.

Reply:
Yeah, I just presumed that we would want the same signature as stan_rdump. I agree that its not intuitive. Will change back, no problem at all! Aslo prefer write_stan_json much better. Again wanted to resemble stan_rdump.

Reply:
Cool. I think once cmdstanr is released  it probably actually makes sense for us to deprecate `rstan::stan_rdump` and put a version of it into cmdstanr called `write_stan_rdump` with the same signature as the new `write_stan_json`. Really the only reason rstan needed to have `stan_rdump` was because an R package for all things cmdstan didn’t exist at the time.